infrastructure/src/main/kotlin/DetektPlugin.kt

```diff
@@ -52,6 +54,26 @@ private fun Project.configureDetektTasks(extension: RedmadrobotExtension) {
     detektTask(extension, "detektAll") {
         description = "Runs over whole code base without the starting overhead for each module."
     }
+
+    if (extension.detekt.checkDiffOnly && extension.detekt.targetDiffBranch.isNotEmpty()) {
```

Comment:
Looks like `checkDiffOnly` can be omitted, `targetDiffBranch.isNotEmpty()` will be enough

Reply:
I changed the parameters for detektDiff task and changed this condition. Now the detektDiff parameters are available only if `checkOnlyDiffWithBranch` specified.